File: api/app/graphql/resolvers/playlists.rb
module Resolvers
  class Playlists < BaseResolver
    type Types::PlaylistType.connection_type, null: false

    argument :query, String, required: false, description: "Search query."

    def resolve(query: "*")
      ::Playlist.search(
        query,
        fields: [:title, :description],
        match: :word_middle,
        misspellings: {below: 5},
        order: {title: :asc},
        includes: [
          image_attachment: :blob
        ]
      ).results
    end
  end
end

File: native/graphql/queries/playlists.ts
import { gql } from '@apollo/client'

export const PLAYLISTS = gql`
	query Playlists {
		playlists {
			edges {
				node {
					id
					title
					description
					image {
						blob {
							url
						}
					}
				}
			}
		}
	}
`
